runtime/tests/runtime_integration_tests/test_reallocate_dpe_context_limits.rs

```diff
@@ -0,0 +1,296 @@
+// Licensed under the Apache-2.0 license.
+
+use crate::common::{execute_dpe_cmd, run_rt_test, DpeResult, RuntimeTestArgs};
+use caliptra_api::mailbox::{
+    CommandId, ReallocateDpeContextLimitsReq, ReallocateDpeContextLimitsResp,
+};
+use caliptra_builder::{
+    firmware::{APP_WITH_UART, FMC_WITH_UART},
+    ImageOptions,
+};
+use caliptra_common::mailbox_api::{MailboxReq, MailboxReqHeader};
+use caliptra_drivers::CaliptraError;
+use caliptra_hw_model::{DefaultHwModel, HwModel, ModelError};
+use dpe::{
+    commands::{Command, DeriveContextCmd, DeriveContextFlags},
+    context::ContextHandle,
+    DPE_PROFILE,
+};
+use zerocopy::FromBytes;
+
+fn fill_max_dpe_contexts(model: &mut DefaultHwModel, pl0_limit: u32, pl1_limit: u32) {
+    const BASE_DERIVE_CONTEXT_CMD: DeriveContextCmd = DeriveContextCmd {
+        handle: ContextHandle::default(),
+        data: [0; DPE_PROFILE.get_tci_size()],
+        flags: DeriveContextFlags::MAKE_DEFAULT,
+        tci_type: 0,
+        target_locality: 0,
+    };
+
+    // 32 contexts = 1 root node (PL0)+
+    //               1 rt_journey (PL0)
+    //               (pl0_limit - 2) PL0 contexts in loop +
+    //               1 PL1 context as transition +
+    //               (pl1_limit - 1) PL1 contexts in loop
+
+    // Fill PL0 contexts
+    for _ in 0..(pl0_limit - 2) {
+        let _ = execute_dpe_cmd(
+            model,
+            &mut Command::DeriveContext(&BASE_DERIVE_CONTEXT_CMD),
+            DpeResult::Success,
+        );
+    }
+
+    // Trigger failure by trying to derive one more context to PL0
+    let _ = execute_dpe_cmd(
+        model,
+        &mut Command::DeriveContext(&BASE_DERIVE_CONTEXT_CMD),
+        DpeResult::MboxCmdFailure(CaliptraError::RUNTIME_PL0_USED_DPE_CONTEXT_THRESHOLD_REACHED),
+    );
+
+    // Fill PL1 contexts
+    if pl1_limit > 0 {
```

Comment:
When pl1_limit is 0, the for loop will be skipped. Should we remove this if statement?